suro-core/src/test/java/com/netflix/suro/sink/TestSinkManager.java

```diff
@@ -94,4 +94,84 @@ public void testSinkOpenFailure() throws Exception {
         Assert.assertEquals(3, sink2.openAttempts.get());
         assertThat(sinkManager.getSink("sink2"), Matchers.<Sink>sameInstance(sink2));
     }
+
+    private static class MockSinkWithBlockingClose implements Sink {
+        private volatile boolean isOpened = false;
+
+        @Override
+        public void writeTo(MessageContainer message) {
+        }
+
+        @Override
+        public void open() {
+            isOpened = true;
+        }
+
+        @Override
+        public void close() {
+            // block close for 1,000 ms
+            try {
+                Thread.sleep(1000);
```

Comment:
Instead of sleep, what about replacing with CountDownLatch? We can unblock through the code rather than depending on the timing schedule.